tests/test_widget_child_moving.py

```diff
@@ -36,28 +36,66 @@ async def test_move_child_not_our_child() -> None:
 async def test_move_child_to_outside() -> None:
     """Test attempting to move relative to a widget that isn't a child."""
     async with App().run_test() as pilot:
-        child = Widget(Widget())
+        child = Widget()
         await pilot.app.mount(child)
         with pytest.raises(WidgetError):
             pilot.app.screen.move_child(child, before=Widget())
 
 
-async def test_move_child_before_itself() -> None:
-    """Test moving a widget before itself."""
+@pytest.mark.parametrize(
+    ["type_of_child", "before"],
+    [
+        ("widget", "widget"),
+        ("widget", "index"),
+        ("index", "widget"),
+        ("index", "index"),
+    ],
+)
+async def test_move_child_before_itself(type_of_child: str, before: str) -> None:
+    """Test moving a widget before itself.
+
+    Regression test for https://github.com/Textualize/textual/issues/1743
+    """
+
+    widget = Widget()
+    keys = {
+        "widget": widget,
+        "index": 0,
+    }
+    child = keys[type_of_child]
+    before = keys[before]
 
     async with App().run_test() as pilot:
-        child = Widget(Widget())
-        await pilot.app.mount(child)
-        pilot.app.screen.move_child(child, before=child)
-
+        await pilot.app.mount(widget)
+        pilot.app.screen.move_child(child, before=before)
+
+
+@pytest.mark.parametrize(
+    ["type_of_child", "after"],
+    [
+        ("widget", "widget"),
+        ("widget", "index"),
+        ("index", "widget"),
+        ("index", "index"),
+    ],
+)
+async def test_move_child_after_itself(type_of_child: str, after: str) -> None:
+    """Test moving a widget after itself.
+
+    Regression test for https://github.com/Textualize/textual/issues/1743
+    """
+
+    widget = Widget()
+    keys = {
+        "widget": widget,
+        "index": 0,
+    }
+    child = keys[type_of_child]
+    after = keys[after]
```

Comment:
Why the extra lookup here? Does something like the code below not work?

```python
[
  (Widget(), Widget()),
  (Widget(), 0),
  # ...
]
```

And just directly pass in `child` and `after`?

Reply:
Your proposal will generate many different `Widget` instances and I need to check what happens when the child and before/after `is` the same instance, whereas if `child, after = Widget(), Widget()`, then we get `child is not after`.

Reply:
I see. The reason that I'm asking is that I find it hard to follow, and the use of conditionals and lookups (which are basically just conditionals) is usually considered a code smell in a parametrised test - it suggests the test is trying to do cover too many things, and there's almost always a better way. Usually, that means breaking the test up into multiple, more readable tests, even if it means duplication.

In this case, I think you're trying to shoehorn in a "a widget can be specified as a integer" test into this test, rather than covering it off in an isolated manner (a separate unit test). If you do this in a separate test, then you can let this test focus on one unit of functionality: move_child nooping.

One other thing is - what is this intended to test? It just calls `move_child` then does nothing. The title of the PR explains that it should no-op, so we could check (within reason) that that's the actual behaviour.

Maybe something like this, to ensure the DOM isn't modified and that it is in fact a null operation?

```python
screen = pilot.app.screen
await pilot.app.mount(widget)
dom_before = screen.walk_children()

screen.move_child(widget, after=widget)

dom_after = screen.walk_children()
assert dom_before == dom_after
```

Reply:
Point taken regarding lookups/conditionals and parametrised tests.

Regarding the testing, I'm testing that the function call doesn't raise an error (before the PR, it did).
I can be extra safe and check that the DOM did stay the same, but given that the screen has a single `Widget` children, doesn't make a lot of sense to me.

Reply:
Thanks for your feedback. I've addressed it in e03de27.